src/adapters/CCIPAdapter.sol

```diff
@@ -65,58 +76,44 @@ contract CCIPAdapter is Auth, ICCIPAdapter {
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc IAdapter
-    function send(uint16 centrifugeId, bytes calldata payload, uint256 gasLimit, address refund)
-    external
-    payable
-    returns (bytes32 adapterData)
+    function send(uint16 centrifugeId, bytes calldata payload, uint256 gasLimit, address)
+        external
+        payable
+        returns (bytes32 adapterData)
     {
         require(msg.sender == address(entrypoint), NotEntrypoint());
         CCIPDestination memory destination = destinations[centrifugeId];
         require(destination.chainSelector != 0, UnknownChainId());
 
-        // Build CCIP message
-        Client.EVM2AnyMessage memory ccipMessage = Client.EVM2AnyMessage({
-            receiver: abi.encode(destination.adapter),
+        IClient.EVM2AnyMessage memory ccipMessage = IClient.EVM2AnyMessage({
+            receiver: abi.encode(destination.addr),
             data: payload,
-            tokenAmounts: new Client.EVMTokenAmount[](0),
+            tokenAmounts: new IClient.EVMTokenAmount[](0),
             feeToken: address(0), // Pay in native token
-            extraArgs: Client._argsToBytes(Client.EVMExtraArgsV1({gasLimit: gasLimit}))
+            extraArgs: _argsToBytes(IClient.EVMExtraArgsV1({gasLimit: gasLimit}))
         });
 
-        // Get exact fee amount needed
```

Comment:
The system always calls `adapter.estimate` first and then passes exactly this as `msg.value` to `adapter.send`. So there will never be any excess gas directly here.

There will often be excess gas on the destination chain as the gas limit is overestimated (significantly).